embeddings/evaluator/text_classification_evaluator.py

```diff
@@ -1,21 +1,44 @@
-from typing import Any, Dict, List, Sequence, Union
+from typing import Any, Dict, List, Type, Union
 
 import torch
 from numpy import typing as nptyping
 
+from embeddings.evaluator.evaluation_results import Predictions, TextClassificationEvaluationResults
 from embeddings.evaluator.metrics_evaluator import MetricsEvaluator
 from embeddings.metric.hugging_face_metric import HuggingFaceMetric
 from embeddings.metric.metric import Metric
 
 
-class TextClassificationEvaluator(MetricsEvaluator):
-    @property
+class TextClassificationEvaluator(MetricsEvaluator[TextClassificationEvaluationResults]):
     def metrics(
         self,
-    ) -> Sequence[Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
-        return [
-            HuggingFaceMetric("accuracy"),
-            HuggingFaceMetric("f1", compute_kwargs={"average": "macro"}),
-            HuggingFaceMetric("recall", compute_kwargs={"average": "macro"}),
-            HuggingFaceMetric("precision", compute_kwargs={"average": "macro"}),
-        ]
+    ) -> Dict[str, Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
+        return {
+            "accuracy": HuggingFaceMetric("accuracy"),
+            "f1_macro": HuggingFaceMetric("f1", compute_kwargs={"average": "macro"}),
+            "f1_micro": HuggingFaceMetric("f1", compute_kwargs={"average": "micro"}),
+            "f1_weighted": HuggingFaceMetric("f1", compute_kwargs={"average": "weighted"}),
+            "recall_macro": HuggingFaceMetric("recall", compute_kwargs={"average": "macro"}),
+            "recall_micro": HuggingFaceMetric("recall", compute_kwargs={"average": "micro"}),
+            "recall_weighted": HuggingFaceMetric("recall", compute_kwargs={"average": "weighted"}),
+            "precision_macro": HuggingFaceMetric("precision", compute_kwargs={"average": "macro"}),
+            "precision_micro": HuggingFaceMetric("precision", compute_kwargs={"average": "micro"}),
+            "precision_weighted": HuggingFaceMetric(
+                "precision", compute_kwargs={"average": "weighted"}
+            ),
+        }
+
+    @property
+    def evaluation_results_cls(self) -> Type[TextClassificationEvaluationResults]:
+        return TextClassificationEvaluationResults
+
+    def evaluate(
+        self, data: Union[Dict[str, nptyping.NDArray[Any]], Predictions]
+    ) -> TextClassificationEvaluationResults:
+        data = Predictions(**data) if isinstance(data, dict) else data
+        result = dict()
+        for field, metric in self.metrics().items():
+            [computed] = metric.compute(y_true=data.y_true, y_pred=data.y_pred).values()
+            result[field] = computed
```

Comment:
```suggestion
            result[field] = metric.compute(y_true=data.y_true, y_pred=data.y_pred).values()[0]
```

Reply:
`[x] = some_list` syntax ensures that there was only one element (raises error if not) which is safer and more readable

Reply:
Nice remark, before that I wasn't aware of such syntax. I checked it during this review :)